silkworm/core/common/lru_cache.hpp

```diff
@@ -51,9 +52,57 @@ class lru_cache {
     typedef typename std::pair<key_t, value_t> key_value_pair_t;
     typedef typename std::list<key_value_pair_t>::iterator list_iterator_t;
 
-    explicit lru_cache(size_t max_size) : _max_size(max_size) {}
-
+    explicit lru_cache(size_t max_size, bool thread_safe = false) : _max_size(max_size), _thread_safe(thread_safe) {}
     void put(const key_t& key, const value_t& value) {
+        if (_thread_safe) {
+            const std::lock_guard<std::mutex> lock(_access);
```

Comment:
You probably need to skip the mutex stuff when compiling for wasm, similar to how it's done in `core/common/object_pool.hpp`, because wasm doesn't support multi-threading (or perhaps it's experimental there).